azure-ipam/ipam.go

```diff
@@ -132,8 +134,12 @@ func (p *IPAMPlugin) CmdDel(args *cniSkel.CmdArgs) error {
 	p.logger.Debug("Making request to CNS")
 	// cnsClient enforces it own timeout
 	if err := p.cnsClient.ReleaseIPAddress(context.TODO(), req); err != nil {
-		p.logger.Error("Failed to release IP address from CNS", zap.Error(err), zap.Any("request", req))
-		return cniTypes.NewError(cniTypes.ErrTryAgainLater, err.Error(), "failed to release IP address from CNS")
+		if errors.Is(err, &cnscli.ConnectionFailureErr{}) {
+			fsnotify.WatcherAddFile(args.ContainerID)
```

Comment:
we still need to do err validation and return failure if not able to update the file